src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -189,15 +189,45 @@ public class BlobFileStore : IFileStore
     }
 
     /// <inheritdoc />
-    public async Task DeleteFileIfExistsAsync(long version, string partitionName, CancellationToken cancellationToken)
+    public async Task DeleteFileIfExistsAsync(long version, Partition partition, FileProperties fileProperties, CancellationToken cancellationToken)
     {
+        EnsureArg.IsNotNull(partition);
 
-        BlockBlobClient blobClient = GetInstanceBlockBlobClient(version, partitionName);
-        _logger.LogInformation("Trying to delete DICOM instance file with watermark '{Version}'.", version);
+        BlockBlobClient blobClient = GetInstanceBlockBlobClient(version, partition, fileProperties);
+        _logger.LogInformation("Trying to delete DICOM instance file with watermark: '{Version}' and PartitionKey: {PartitionKey}.", version, partition.Key);
 
         EmitTelemetry(nameof(DeleteFileIfExistsAsync), OperationType.Input);
 
-        await ExecuteAsync(() => blobClient.DeleteIfExistsAsync(DeleteSnapshotsOption.IncludeSnapshots, conditions: null, cancellationToken));
+        await ExecuteAsync(async () =>
+        {
+            try
+            {
+                bool exists = await blobClient.ExistsAsync(cancellationToken);
```

Comment:
I think you don't have to do 2 calls. DeleteExistsAsync `Returns true if container exists and was marked for deletion, return false otherwise.`

Reply:
What exactly happens in this case? I assume you get a `412` back, right? Is there a separate error code or inner exception we could use to differentiate? I ask because if we check and delete in two separate calls, there is always the chance something happened in between

Reply:
In fact -- do we care why it failed outside of logs? If it does fail for either reason, we can just query the state of the blob afterwards, right?

Reply:
Sure, but the message can be pretty straight-forward if we don't check. That would significantly simplify this and remove any race conditions.

```
"There was a problem deleting the DICOM file with SOP instance ID foo. Expected the file to be at '/foo/bar/baz.dcm' with eTag 0x12345, but the file has since been changed or deleted."
```